### PR TITLE
[WEBSITE-759] Automated Browser testing

### PR DESCRIPTION
**Pull request by naseberry**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+SAUCE_USERNAME?=unknown
+SAUCE_ACCESS_KEY?=unknown
+
+# Node module variables
+MOCHA=./node_modules/.bin/mocha
+MOCHA_PARALLEL=./node_modules/.bin/mocha-parallel-tests
+
+
+# Concurrent test
+sauce:
```
Comment (jakemulley) [inline Makefile:14]: This should be a .PHONY target (https://www.gnu.org/software/make/manual/html_node/Phony-Targets.html)

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+	@env URL=$(URL) SAUCE_USERNAME=$(SAUCE_USERNAME) SAUCE_ACCESS_KEY=$(SAUCE_ACCESS_KEY) $(MOCHA_PARALLEL) test/
+
+
+# Incremental test
+isauce:
```
Comment (jakemulley) [inline Makefile:19]: This should be a .PHONY target (https://www.gnu.org/software/make/manual/html_node/Phony-Targets.html)

Comment (jakemulley) [reply Makefile:19]: It might also be better to give it a more meaningful name than `isauce`

File: test/index.js [hunk 3]
```diff
@@ -0,0 +1,180 @@
+/**
+ * Browser testing
+ */
+
+
+var fs = require('fs'),
```
Comment (jakemulley) [inline test/index.js]: Tabbing here does not follow the `.editorconfig` specification

File: test/index.js [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+			'name': 'Desktop-WIN-Firefox-51',
+			'browserName': 'firefox',
+			'platform': 'Windows 10',
+			'version': '51'
+		},
+
+		/* TODO: Mobile
+		 * https://wiki.saucelabs.com/display/DOCS/Common+Error+Messages#CommonErrorMessages-NewSessionRequestwasCancelledbeforeaSauceLabsVirtualMachinewasFound
+		{
+			'name': 'Mobile-iOS-safari-9',
```
Comment (jakemulley) [inline test/index.js]: Do we want to commit massive TODO comments?

File: test/index.js [hunk 5]
```diff
@@ -0,0 +1,180 @@
...
+	i++;
+	return fs.writeFileSync('./screenshot/' + name + '-'+ i + '.png', data, 'base64');
+}
+
+
+function ukp_takeScreenshot(name) { // Selenium, take a screenshot
+	return driver.takeScreenshot().then(function(data) {
+		return ukp_writeFile(data, name);
+	});
+}
```
Comment (jakemulley) [inline test/index.js:179]: This might be better restructured to use Promises

**Review by jakemulley — CHANGES_REQUESTED** [review]

File: test/index.js (hunk 5 above)
Comment (UsmanAfzal) [reply test/index.js:179]: Possibly. That would be better as another commit that can be reviewed to compare

**Review by jakemulley — APPROVED** [review]

**Pull request merged**